pkg/performanceprofile/controller/performanceprofile_controller_hypershift.go

```diff
@@ -0,0 +1,650 @@
+			},
+		},
+		Data: map[string]string{
+			mcoConfigMapConfigKey: machineconfigManifest,
+		},
+	}
+}
+
+func KubeletConfigConfigMap(owner *corev1.ConfigMap, performanceProfileName string, nodePoolNamespacedName string, kubeletconfigManifest string) *corev1.ConfigMap {
+	return &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: owner.GetNamespace(),
+			Name:      fmt.Sprintf("kc-%s", performanceProfileName),
+			Labels: map[string]string{
+				ntoGeneratedMachineConfigLabel:        "true",
+				hypershiftPerformanceProfileNameLabel: performanceProfileName,
+				hypershiftNodePoolLabel:               parseNamespacedName(nodePoolNamespacedName),
+			},
+			Annotations: map[string]string{
+				hypershiftNodePoolLabel: nodePoolNamespacedName,
+			},
+			OwnerReferences: []metav1.OwnerReference{
+				newOwnerReference(owner, owner.GroupVersionKind()),
+			},
+		},
+		Data: map[string]string{
+			mcoConfigMapConfigKey: kubeletconfigManifest,
+		},
+	}
+}
+
+func newOwnerReference(owner metav1.Object, gvk schema.GroupVersionKind) metav1.OwnerReference {
+	blockOwnerDeletion := false
+	isController := false
+	return metav1.OwnerReference{
+		APIVersion:         gvk.GroupVersion().String(),
+		Kind:               gvk.Kind,
+		Name:               owner.GetName(),
+		UID:                owner.GetUID(),
+		BlockOwnerDeletion: &blockOwnerDeletion,
+		Controller:         &isController,
+	}
+}
+func convertMachineConfig(origMC *mcov1.MachineConfig) *mcfgv1.MachineConfig {
+
+	return &mcfgv1.MachineConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: mcfgv1.SchemeGroupVersion.String(),
+			Kind:       "MachineConfig",
+		},
+		ObjectMeta: origMC.ObjectMeta,
+		// althoug mco.MC.Spec and hypershift mco.MC.Spec are defined in
+		// different files they have the same structure, so the conversion is
+		// direct
+		//TODO - Is there any way to do this conversion in a type safe manner?
+		Spec: mcfgv1.MachineConfigSpec(origMC.Spec),
+	}
+}
+
+func convertKubeletConfig(origKC *mcov1.KubeletConfig) *mcfgv1.KubeletConfig {
+
+	return &mcfgv1.KubeletConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: mcfgv1.SchemeGroupVersion.String(),
+			Kind:       "KubeletConfig",
+		},
+		ObjectMeta: origKC.ObjectMeta,
+		//TODO - Is there any way to do this conversion in a type safe manner?
+		//NOTE - MachineConfigPoolSelector left empty as NodePool is the one
+		// that relates nodes with MachineConfigs in hypershift.
+		Spec: mcfgv1.KubeletConfigSpec{
+			KubeletConfig: origKC.Spec.KubeletConfig,
+		},
+	}
+}
+
+// parseManifests parses a YAML or JSON document that may contain one or more
+// kubernetes resources.
+func parsePerformanceProfileManifest(data []byte, nodePoolName string) (*performancev2.PerformanceProfile, error) {
+	scheme := runtime.NewScheme()
+	performancev2.AddToScheme(scheme)
+
+	//REVIEW - As serializer is used many times it could be worthy to create it just one time and keep it somewhere
```

Comment:
Maybe this could be moved to the shared hypershift pkg, and instantiated in init(). We would need to add multiple types to the scheme, but I think this is safe.